### PR TITLE
Media and Log functions

### PR DESCRIPTION
**Pull request by maximvelichko**
1) @jessbo's changes to retrieve media files (https://github.com/jessbo/python-amcrest)
2) Log retrieval functionality

### CODE REVIEW

File: src/amcrest/media.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+#
+# vim:sw=4:ts=4:et
+
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Media(object):
+
```
[the remark this answers is not in the document]
Comment (maximvelichko) [reply src/amcrest/media.py]: I am not sure what is it complaining about...

**Comment by maximvelichko** [comment]
I believe the code coverage is based on test in tests/test_utils.py
They are for amcrest.utils only, so I see no way to increase the coverage without introducing e.g. mocks for src/amcrest/http.py. (And any added functionality would decrease the coverage).

If I miss something, and there is a way to increase coverage for what I am adding - please let me know, I'd be glad to add it

File: .travis.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ matrix:
       env: TOXENV=py35
     - python: "3.6"
       env: TOXENV=py36
-    - python: "3.4.2"
+    - python: "3.4"
```
Comment (tchellomello) [inline .travis.yml]: Let's make it to python 3.7
```yaml
    - python: "3.7"
      env: TOXENV=py37
      dist: xenial
```

File: src/amcrest/log.py [hunk 3]
```diff
@@ -27,3 +27,42 @@ def log_show(self, start_time, end_time):
...
+            'log.cgi?action=startFind'
+            '&condition.StartTime={0}&condition.EndTime={1}'
+            .format(
+                start_time.strftime('%Y-%m-%d %H:%M:%S'),
+                end_time.strftime('%Y-%m-%d %H:%M:%S')))
+
+        return ret.content.decode('utf-8')
+
+    def log_find_next(self, token):
+        ret = self.command('log.cgi?action=doFind&token={0}&count=100'
```
Comment (tchellomello) [inline src/amcrest/log.py]: @maximvelichko the parameter `count=100` could be passed as a parameter into this method? So this way the user could modify it or is it a hardcoded value that you found to work better?

File: src/amcrest/media.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tchellomello) [reply src/amcrest/media.py]: @maximvelichko https://pypi.org/project/flake8-black/

File: src/amcrest/media.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                with all the flags.
+
+        event : by which event the record file is triggered. It is an array.
+                The index starts from 0. The range of event is {"AlarmLocal",
+                "VideoMotion", "VideoLoss", "VideoBlind", "Traffic*"}. This
+                condition can be omitted. If omitted, find files of all the
+                events. stream : which video stream type you want to find.
+                The range of stream is {"Main", "Extra1", "Extra2", "Extra3"}.
+                If omitted, find files with all the stream types.
+        """
```
Comment (tchellomello) [inline src/amcrest/media.py]: Awesome documentation!!!!! :1st_place_medal:

File: src/amcrest/media.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
```
Comment (tchellomello) [inline src/amcrest/media.py]: Another great documentation!!

File: src/amcrest/media.py [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+            'mediaFileFind.cgi?action=findFile&object={0}&condition.Channel'
+            '={1}&condition.StartTime={2}&condition.EndTime={3}{4}{5}{6}{7}{8}'
+            .format(factory_id, channel, start_time, end_time, c_dirs, c_types,
+                    c_flag, c_events, c_vs)
+        )
+        return ret.content.decode('utf-8')
+
+    def media_file_find_next(self, factory_id):
+        ret = self.command(
+            'mediaFileFind.cgi?action=findNextFile&object={0}&count=100'
```
Comment (tchellomello) [inline src/amcrest/media.py]: Same as before. Could you make this as a parameter as mentioned above?

File: src/amcrest/media.py (hunk 4 above)
Comment (maximvelichko) [reply src/amcrest/media.py]: Kudos to @jessbo, I've just kept it ;-)

Comment (tchellomello) [reply src/amcrest/media.py]: @jessbo congrats!

**Pull request merged**